### PR TITLE
Added jax.numpy.product()

### LINKED ISSUE #14485: product

### PR DESCRIPTION
**Pull request by KevinUli**
close #14485

### CODE REVIEW

File: ivy/functional/frontends/jax/numpy/mathematical_functions.py [hunk 1]
```diff
@@ -606,3 +606,28 @@ def polymul(a1, a2, *, trim_leading_zeros=False):
...
+
+@to_ivy_arrays_and_back
+def product(a,
+            *,
+            axis=None,
+            dtype=None,
+            keepdims=False,
+            initial=None,
+            where=None,
+            promote_integer=True,
```
Comment (hmahmood24) [inline ivy/functional/frontends/jax/numpy/mathematical_functions.py:618]: The original `jax` function has the argument `promote_integers`

File: ivy_tests/test_ivy/test_frontends/test_jax/test_jax_numpy_math.py [hunk 2]
```diff
@@ -2773,3 +2799,41 @@ def test_jax_numpy_polymul(
...
+        keepdims,
+        initial,
+        frontend,
+        test_flags,
+        fn_tree,
+        on_device
+):
+    input_dtypes, x, axis, dtype, where = dtype_x_axis_dtype_where
+    if ivy.current_backend_str() == "torch":
+        assume(not test_flags.as_variable[0])
```
Comment (hmahmood24) [inline ivy_tests/test_ivy/test_frontends/test_jax/test_jax_numpy_math.py:2821]: It is generally not recommended to use `assume` inside the test function itself unless absolutely necessary. Why have we included this check here especially for the `torch` backend?

**Review by hmahmood24 — CHANGES_REQUESTED** [review]
Hey @KevinUli, thanks so much for this PR. I have left a few comments for you to address. Additionally, the [tests](https://github.com/unifyai/ivy/actions/runs/4774413312/jobs/8488074818?pr=14490#step:3:192) seem to be failing for `numpy` and I can also see some [linting/formatting issues](https://github.com/unifyai/ivy/actions/runs/4774413311/jobs/8488074752?pr=14490#step:5:15) with the PR. 

Can you kindly resolve these changes/observations and ask for a re-review from me? Thanks and good luck! 😃

File: ivy/functional/frontends/jax/numpy/mathematical_functions.py (hunk 1 above)
Comment (KevinUli) [reply ivy/functional/frontends/jax/numpy/mathematical_functions.py]: I have changed `promote_integer` to `promote_integers`

File: ivy_tests/test_ivy/test_frontends/test_jax/test_jax_numpy_math.py (hunk 2 above)
Comment (KevinUli) [reply ivy_tests/test_ivy/test_frontends/test_jax/test_jax_numpy_math.py:2765]: The current `torch` backend does not seem to support this. The numpy.prod test seem to also exclude `torch` backend.

**Comment by KevinUli** [comment]
The remaining failing checks for lint is not from my commits

File: ivy_tests/test_ivy/test_frontends/test_jax/test_jax_numpy_math.py (hunk 2 above)
Comment (KevinUli) [reply ivy_tests/test_ivy/test_frontends/test_jax/test_jax_numpy_math.py:2764]: Should I leave it as a failing check instead?

Comment (hmahmood24) [reply ivy_tests/test_ivy/test_frontends/test_jax/test_jax_numpy_math.py:2820]: Right. Yeah sure no worries then. Thanks!

**Comment by hmahmood24** [comment]
@KevinUli Thanks a lot for the changes. Everything looks good to merge! Thanks so much for the contribution! 🚀

**Review by hmahmood24 — APPROVED** [review]

**Pull request merged**